base/compiler/inferencestate.jl

```diff
@@ -661,16 +661,20 @@ mutable struct IRInterpretationState
     parent # ::Union{Nothing,AbsIntState}
 
     function IRInterpretationState(interp::AbstractInterpreter,
-        method_info::MethodInfo, ir::IRCode, mi::MethodInstance, argtypes::Vector{Any},
+        method_info::MethodInfo, ir::IRCode, mi::Union{Nothing, MethodInstance}, argtypes::Vector{Any},
         world::UInt, min_world::UInt, max_world::UInt)
         curridx = 1
         given_argtypes = Vector{Any}(undef, length(argtypes))
         for i = 1:length(given_argtypes)
             given_argtypes[i] = widenslotwrapper(argtypes[i])
         end
-        given_argtypes = va_process_argtypes(optimizer_lattice(interp), given_argtypes, mi)
-        argtypes_refined = Bool[!⊑(optimizer_lattice(interp), ir.argtypes[i], given_argtypes[i])
-            for i = 1:length(given_argtypes)]
+        if mi === nothing
+            argtypes_refined = zeros(Bool, length(argtypes))
+        else
+            given_argtypes = va_process_argtypes(optimizer_lattice(interp), given_argtypes, mi)
+            argtypes_refined = Bool[!⊑(optimizer_lattice(interp), ir.argtypes[i], given_argtypes[i])
+                for i = 1:length(given_argtypes)]
+        end
```

Comment:
Can't we pass a "original" `mi` here and configure this behavior via a keyword argument?
`frame_instance` is an essential part of `AbsIntState` and used e.g. as a target for storing backedges.

Reply:
That was kinda my original PR that just passed the mismatched MI and added a kwarg instead of the `mi === nothing` line basically. But then Keno suggested making MI a maybe

Reply:
The scenario here is that we don't have a method that represent this piece of code because we're conjuring up IR. However, upon further reflection, we already have this scenario for toplevel code that we're infering elsewhere and what we're doing there is creating a MethodInstance without a Method, so we should probably do the same here then.